src/main.rs

```diff
@@ -166,11 +192,14 @@ fn check_path_separator_length(path_separator: Option<&str>) -> Result<()> {
     }
 }
 
-fn construct_config(mut opts: Opts, pattern_regex: &str) -> Result<Config> {
+fn construct_config(mut opts: Opts, pattern_regexps: &Vec<String>) -> Result<Config> {
```

Comment:
This should probably be:

```suggestion
fn construct_config(mut opts: Opts, pattern_regexps: &[String]) -> Result<Config> {
```
(note that a `Vec<String>` will implicitly dereference to a `&[String]`, so `&pattern_regexps` would still fit the type)

We could almost get away with just accepting an iterator, except that we would need to be able to iterate over it it twice in order to determine if we should be case sensitive or not if the options aren't expllicit.